### PR TITLE
Silencing actions

### PR DESCRIPTION
**Pull request by RoiGlinik**
add list and delete silence actions

### CODE REVIEW

File: src/robusta/core/sinks/opsgenie/opsgenie_sink.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from gettext import find
```
Comment (aantn) [inline src/robusta/core/sinks/opsgenie/opsgenie_sink.py:1]: Unused import

File: playbooks/robusta_playbooks/silence.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        return
+
+    response = requests.get(
+        f"{alertmanager_url}/api/v2/silences",
+        headers={"Content-type": "application/json"},
+    )
+
+    response.raise_for_status()
+
+    silence_list = [list(Silence(**silence).list()) for silence in response.json()]
```
Comment (aantn) [inline playbooks/robusta_playbooks/silence.py:87]: Do we need the call to `list()`? Isn't it already a list?

**Review by aantn — COMMENTED** [review]
I left some comments - don't think there are any actual bugs there, mostly just aesthetics.

In general, do we need to send TableBlocks() as the playbook result? If the UI is just parsing the data and we're never interpreting this as a real table (for display) you can use a JsonBlock too.

**Comment by arikalon1** [comment]
> I left some comments - don't think there are any actual bugs there, mostly just aesthetics.
> 
> In general, do we need to send TableBlocks() as the playbook result? If the UI is just parsing the data and we're never interpreting this as a real table (for display) you can use a JsonBlock too.

I recommended to use that, so this playbook result will be able to be used by other sinks as well, with no additional work.
JsonBlock isn't supported in most sinks.

File: playbooks/robusta_playbooks/silence.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+from robusta.api import *
+
+
+class Matcher(BaseModel):
+    isEqual: bool
+    isRegex: bool
+    name: str
+    value: str
+
+    def stringify(self) -> str:
```
[the remark this answers is not in the document]
Comment (RoiGlinik) [reply playbooks/robusta_playbooks/silence.py:11]: this was actually not necessary anymore . but good point anyhow

File: src/robusta/core/reporting/base.py [hunk 4]
```diff
@@ -146,3 +149,20 @@ def add_enrichment(self, enrichment_blocks: List[BaseBlock], annotations=None):
...
+        labels["alertname"] = self.aggregation_key
+        labels["cluster"] = cluster_id
+        if self.subject.namespace:
+            labels["namespace"] = self.subject.namespace
+
+        kind : str = self.subject.subject_type.value
+        if kind and self.subject.name:
+            labels[kind] = self.subject.name
+ 
+        #todo add more resources.
```
[the remark this answers is not in the document]
Comment (RoiGlinik) [reply src/robusta/core/reporting/base.py:164]: its for future to add other labels . we added some of the more solid labels

**Pull request merged**